packages/provider/src/tasks/calculateSolutions.ts

```diff
@@ -0,0 +1,84 @@
+import { CaptchaStates } from '@prosopo/types'
+import { ProsopoEnvError, logger } from '@prosopo/common'
+import { ProviderEnvironment } from '@prosopo/types-env'
+import { ScheduledTaskNames } from '@prosopo/types'
+import { Tasks } from './tasks'
+import { calculateNewSolutions, checkIfTaskIsRunning, updateSolutions } from '../util'
+import { captchaSort } from '@prosopo/datasets'
+export class CalculateSolutionsTask extends Tasks {
+    constructor(env: ProviderEnvironment) {
+        super(env)
+        this.logger = logger(env.config.logLevel, 'CalculateSolutionsTask')
+    }
+
+    /**
+     * Apply new captcha solutions to captcha dataset and recalculate merkle tree
+     */
+    async run(): Promise<number> {
```

Comment:
feels like a pretty big function, but also it's pretty clear what it does. not sure if worth breaking down into smaller methods or not

Reply:
It's going to be separated into a completely different package for downstream processing so probably best to leave refactoring till then.